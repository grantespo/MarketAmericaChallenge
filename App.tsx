// App.tsx
import React from "react";
import { StatusBar } from "expo-status-bar";
import RootNavigationStack from "./navigation/RootNavigationStack";
import { CartProvider } from "./contexts/CartProvider";

const App = () => (
  <>
    <CartProvider>
      <RootNavigationStack />
    </CartProvider>
    <StatusBar style="auto" />
  </>
);

export default App;
