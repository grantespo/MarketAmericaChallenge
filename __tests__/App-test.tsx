import React from 'react';
import { render } from '@testing-library/react-native';
import App from '../App';

jest.mock('../navigation/RootNavigationStack', () => {
  const { Text } = require('react-native');
  return () => <Text>RootNavigationStack</Text>;
});

describe('App component', () => {
  it('renders correctly without errors', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
