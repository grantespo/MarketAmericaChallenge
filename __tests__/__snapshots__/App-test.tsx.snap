// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly without errors 1`] = `
<Text>
  RootNavigationStack
</Text>
`;
